In [7]:
# Dependencies
import pandas as pd
import plotly.express as plt
import os
import statsmodels.api as sm
from sklearn.preprocessing import StandardScaler

In [8]:
bikecounts_file = os.path.join('../','Resources', 'bikecounts.csv')
bikecounts_df = pd.read_csv(bikecounts_file)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

  Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0          0.01           1704.0              3126               4115.0   
1          0.15            827.0              1646               2565.0   
2          0.09            526.0              1232               1695.0   
3      0.47 (S)            521.0              1067               1440.0   
4             0           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [9]:
def remove_unwanted(x):
    for unwanted_char in ['T', '(S)', ' ']:
        x=x.replace(unwanted_char, '')
    return x
bikecounts_df['Precipitation']=bikecounts_df['Precipitation'].apply(remove_unwanted)
bikecounts_df['Precipitation']=bikecounts_df['Precipitation'].replace('',0).astype('float64')
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

   Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0           0.01           1704.0              3126               4115.0   
1           0.15            827.0              1646               2565.0   
2           0.09            526.0              1232               1695.0   
3           0.47            521.0              1067               1440.0   
4           0.00           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [10]:
# Select your independent X terms, and your dependent y term. 
X = bikecounts_df[['Precipitation','High Temp (°F)']]
y = bikecounts_df['Brooklyn Bridge']
X = sm.add_constant(X)
model = sm.OLS(y, X)
results = model.fit()

In [11]:
# Print the linear regression results
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.681
Model:                            OLS   Adj. R-squared:                  0.657
Method:                 Least Squares   F-statistic:                     28.81
Date:                Fri, 08 Oct 2021   Prob (F-statistic):           2.01e-07
Time:                        17:47:49   Log-Likelihood:                -232.03
No. Observations:                  30   AIC:                             470.1
Df Residuals:                      27   BIC:                             474.3
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const           -746.2092    647.158     -1.

In [12]:
coef = results.params
print(f"The equation of our line is y = (Precipitation*{coef[1]}) + (High Temp (°F) *{coef[2]})+ {coef[0]}")

The equation of our line is y = (Precipitation*-3751.0121222347134) + (High Temp (°F) *53.02320134146557)+ -746.2092362024298


* Hypothesis - If we assume that Precipitation and High Temp (°F) has an effect on the number of bikes crossing Brooklyn Bridge then less bikes will cross Brooklyn Bridge.

* Null Hypothesis - If we assume that Precipitation and High Temp (°F) has an effect on the number of bikes crossing Brooklyn Bridge then there is not change.

* R-Squared value of 0.681 would indicate that about 70% of the variance of the dependent variable being studied is explained by the variance of the independent variable